scripts/postlink/ios/postlink.js

```diff
@@ -94,3 +98,42 @@ function findFileByAppName(array, appName) {
 
     return null;
 }
+
+function getDefaultPlistPath() {
+    //this is old logic in case we are unable to find PLIST from xcode/pbxproj - at least we can fallback to default solution
+    return glob.sync(`**/${package.name}/*Info.plist`, ignoreNodeModules)[0];
+}
+
+function getPlistPath(){
+    var xcodeProjectPaths = glob.sync(`**/*.xcodeproj/project.pbxproj`, ignoreNodeModules);
+    if (!xcodeProjectPaths){
+        return getDefaultPlistPath();
+    }
+
+    if (xcodeProjectPaths.length !== 1) {
+        console.log('Could not determine correct xcode proj path to retrieve related plist file, there are multiple xcodeproj under the solution.');
+        return getDefaultPlistPath();
+    }
+
+    var xcodeProjectPath = xcodeProjectPaths[0];
+    var parsedXCodeProj;
+
+    try {
+        var proj = xcode.project(xcodeProjectPath);      
+        //use sync version because there are some problems with async version of xcode lib as of current version
+        parsedXCodeProj = proj.parseSync();
+    }
+    catch(e) {
+        console.log('Couldn\'t read info.plist path from xcode project - error: ' + e.message);
+        return getDefaultPlistPath();
+    }
+
+    var INFO_PLIST_PROJECT_KEY = 'INFOPLIST_FILE';
+    var plistPathValue = parsedXCodeProj.getBuildProperty(INFO_PLIST_PROJECT_KEY);
```

Comment:
@silhouettes, you are absolutely right, this way we cannot guarantee that correct plist file will be taken (although in our case for CodePushDemoApp this will still work because this will take not the first item found but the last one - since there is no "break" in the loop when correct item is found).

Anyway seems that this problem also exists for current solution (we are just taking the first plist file found). So we can try to use more robust solution, where we will look for plist path only in those sections where we have PRODUCT_NAME property defined and matching to package.name. This will work for the most of projects based on default template generated with init command. In case we cannot find this section in pbxproj file we can always fall back to the default solution.

What do you think about this?